### PR TITLE
Test template

### PR DESCRIPTION
**Pull request by phracek**
Test suite for modules and containers.

I would be glad for review and comments.

### CODE REVIEW

**Comment by TomasTomecek** [comment]
LGTM

how do I trigger the sanity tests?

**Comment by jscotka** [comment]
@TomasTomecek  It is very good question how to do it locally.
It is very dependent how you locally build docker container or another final artifacts.
for example 
if you will build local image like: 
```
docker build -t docker-whale .
```
then you can call tests like:
```
URL="docker=docker-whale" make test
```

when I've checked whole Makefile, there could probably be fixed, that docker build uses $(IMAGE_NAME) variable, so that after that build test can use it.
for example (but you have to ensure that you does not have overwritten this variable inside like in run_tests.sh or similar)

File: tests/Makefile [hunk 1]
```diff
@@ -0,0 +1,10 @@
+CMD=python -m avocado run --filter-by-tags=-WIP *.py *.sh
+
+#
+check-docker:
+    MODULE=docker $(CMD)
```
Comment (jscotka) [inline tests/Makefile]: MODULE=docker   should not be defined here I think but in upper level Makefile

File: Makefile [hunk 2]
```diff
@@ -18,4 +18,5 @@ run: build
 	docker run -d $(IMAGE_NAME):x.z
 
 test:
-	run_test.sh
+	cd tests; make all
```
Comment (jscotka) [inline Makefile]: there could be defined like:
```
cd tests; MODULE=docker URL="docker=$(IMAGE_NAME)" make all
```

and I forgot that in that case test: should have contain target `build`

**Pull request merged**